cg/meta/store/base.py

```diff
@@ -29,26 +42,114 @@ def add_new_analysis(bundle_data, case_obj, status, version_obj):
 
 def reset_case_action(case_obj):
     """ Resets action on case """
+
     case_obj.action = None
 
 
-def get_files(deliverables: dict, pipeline: str) -> dict:
+def get_files(deliverables: dict, pipeline: str) -> list:
     """Get all deliverable files from the pipeline"""
 
-    data = [
-        {"path": file["path"], "tags": get_tags(file, pipeline), "archive": False}
+    pipeline_tags = HK_TAGS[pipeline]
+    files = _get_files_non_index(deliverables, pipeline_tags)
+    _check_mandatory_tags(files, PIPELINE_TAGS[pipeline])
+    _convert_tags(files, PIPELINE_TAGS[pipeline], "tags")
+    index_files = _get_files_index(deliverables, pipeline_tags)
+    _convert_tags(index_files, PIPELINE_TAGS[pipeline], "index_tags")
+
+    files.extend(index_files)
+
+    return files
+
+
+def _get_files_non_index(deliverables: dict, pipeline_tags: list) -> list:
+    """ Get all files that are not index files from the deliverables file """
+
+    return [
+        {"path": file["path"], "tags": get_tags(file, pipeline_tags), "archive": False}
         for file in deliverables["files"]
     ]
 
-    return data
+
+def _get_files_index(deliverables: dict, pipeline_tags: list) -> list:
+    """ Get all index files from the deliverables file """
+
+    return [
+        {"path": file["path_index"], "tags": get_tags(file, pipeline_tags), "archive": False}
+        for file in deliverables["files"]
+        if file["path_index"]
+    ]
 
 
-def get_tags(file: dict, pipeline: str) -> list:
+def get_tags(file: dict, pipeline_tags: list) -> list:
     """Get all tags for a file"""
 
-    all_tags = [file["format"], file["id"], file["step"], file["tag"], pipeline]
+    all_tags = [file["format"], file["id"], file["step"], file["tag"]]
+    all_tags.extend(pipeline_tags)
     unique_tags = set(all_tags)
     only_existing_tags = unique_tags - set([None])
     sorted_tags = sorted(list(only_existing_tags))
 
     return sorted_tags
+
+
+def build_bundle(config_data: dict, analysisinfo_data: dict, deliverables: dict) -> dict:
+    """Create a new bundle to store in Housekeeper"""
+
+    pipeline = config_data["samples"][0]["type"]
```

Comment:
Here it would be great with a get_analysis_type def to make sure that a config data is set properly upstream in the parse_config part that is pipeline specific. I also think you mean analysis_type and not pipeline.

Reply:
Changed the var name to analysis_type.

Note that I'm getting the analysis type per sample from the config file in cg.meta.store.mip.parse_config, which calls cg.meta.store.mip._get_sample_analysis_type. In build_bundle I'm simply picking the `type` of the first sample. Is that okay?